spec/bin/synapse_spec.rb

```diff
@@ -0,0 +1,40 @@
+require 'spec_helper'
+
+describe 'parseconfig' do
+  it 'parses a templated config' do
+    allow_any_instance_of(Synapse::Synapse).to receive(:run).and_return(nil)
+    expect(Synapse::Synapse).to receive(:new).
+      with(hash_including(
+       {"services" =>
+          {"test" =>
+            {"default_servers" =>
+              [{"name" => "default1", "host" => "localhost", "port" => 8080}],
+             "discovery" =>
+              {"method" => "zookeeper",
+               "path" => "/airbnb/service/logging/event_collector",
+               "hosts" => ["localhost:2181"],
+               "label_filters" =>
+                [{"label" => "tag", "value" => "config value", "condition" => "equals"}]},
+             "haproxy" =>
+              {"port" => 3219,
+               "bind_address" => "localhost",
+               "server_options" => ["some_haproxy_server_option"],
+               "listen" => ["some_haproxy_listen_option"]}}},
+         "haproxy" =>
+          {"reload_command" => "sudo service haproxy reload",
+           "config_file_path" => "/etc/haproxy/haproxy.cfg",
+           "do_writes" => false,
+           "do_reloads" => false,
+           "do_socket" => false,
+           "global" => ["global_test_option"],
+           "defaults" => ["default_test_option"]},
+         "file_output" => {"output_directory" => "/tmp/synapse_file_output_test"}}
+      )).and_call_original
+    ENV['SYNAPSE_CONFIG'] = "#{File.dirname(__FILE__)}/../support/minimum.conf.yaml"
+    ENV['SYNAPSE_CONFIG_VALUE'] = 'config value'
```

Comment:
reset ENV variables after the test, otherwise they will not be cleaned up.
2 options:
1) set original ENV var to a variable, ie: `cached_env_synapse_config`, assign it back after
2) stub the call to ENV, ie: `allow(ENV).to receive(:[]).with('SYNAPSE_CONFIG').and_return(....)`